src/pages/api/pdf/certificate/[donationId].tsx

```diff
@@ -1,28 +1,29 @@
 import { renderToStream } from '@react-pdf/renderer'
 import { NextApiHandler, NextApiRequest, NextApiResponse } from 'next'
 
-import { PersonResponse } from 'gql/person'
 import { apiClient } from 'service/apiClient'
-import { DonationResponse } from 'gql/donations'
+import { UserDonationResponse } from 'gql/donations'
 import { endpoints } from 'service/apiEndpoints'
 import Certificate from 'components/pdf/Certificate'
+import { authConfig } from 'service/restRequests'
+import { getToken } from 'next-auth/jwt'
 
 const Handler: NextApiHandler = async (req: NextApiRequest, res: NextApiResponse) => {
   const id = Array.isArray(req.query.donationId) ? req.query.donationId[0] : req.query.donationId
 
-  const { data: donation } = await apiClient.get<DonationResponse>(
-    endpoints.donation.getDonation(id).url,
+  const jwt = await getToken({ req })
+  const { data: donation } = await apiClient.get<UserDonationResponse>(
+    endpoints.donation.getUserDonation(id).url,
+    authConfig(jwt?.accessToken),
   )
 
-  let person: PersonResponse | undefined = undefined
-  if (donation?.personId) {
-    const { data } = await apiClient.get<PersonResponse>(
-      endpoints.person.viewPerson(donation.personId).url,
-    )
-    person = data
+  if (!donation) {
+    res.send('What should happen?')
```

Comment:
What should happen in this case? Donation could not be find if is not owned by logged user or if it's for draft donation or it's just invalid. 
Is there a page that should be returned?